### ISSUE TITLE
Callbacks do not work with named arguments

### ISSUE THREAD

**Issue by Nithanaroy** [GoogleChromeLabs/comlink#420]
On Chrome 79, the below implementation which passes callback as a named argument, does not work. Am I missing something?

**index.html**
```html
<!DOCTYPE html>

<script type="module">
  import * as Comlink from "https://unpkg.com/comlink@alpha/dist/esm/comlink.mjs";
  // import * as Comlink from "../../../dist/esm/comlink.mjs";
  
  function callback(value) {
    console.log(`Main from Worker: Result: ${value}`);
  }
  
  async function init() {
    const remoteFunction = Comlink.wrap(new Worker("worker.js"));
    console.log("Main: Call remote func with no args");
    await remoteFunction();

    console.log("Main: Call remote func with copied arg only");
    await remoteFunction(100);

    try {
      console.log("Main: Call remote func with function pointer only");
      await remoteFunction(Comlink.proxy(callback));
    } catch (error) {
      console.error(error)      
    }
    
    try {
      console.log("Main: Call remote func with all args");
      await remoteFunction(100, Comlink.proxy(callback));
    } catch (error) {
      console.error(error)
    }
    
    try {
      console.log("Main: Call remote func with all args as named args");
      await remoteFunction({val: 100, cb: Comlink.proxy(callback)});
    } catch (error) {
      console.error(error)
    }
  }
  init();
</script>
```

**worker.js**
```js
importScripts("https://unpkg.com/comlink@alpha/dist/umd/comlink.js");
// importScripts("../../../dist/umd/comlink.js");

function remoteFunction({ val = 10, cb = s => s } = {}) {
    console.log(`Worker: Caller passed ${val}`)
    cb("A string from a worker");
}

/************************
* Callback is called as expected in the below implementation
**************************/
// function remoteFunction(val = 10, cb = s => s) {
//     console.log(`Worker: Caller passed ${val}`)
//     cb("A string from a worker");
// }

Comlink.expose(remoteFunction);
```

**Comment by surma** [comment]
So all of these tests _except the last one_ work as expected _if_ you have the correct `remoteFunction` implementation. So I am gonna assume that your problem lies with this test only:

```js
 try {
      console.log("Main: Call remote func with all args as named args");
      await remoteFunction({val: 100, cb: Comlink.proxy(callback)});
    } catch (error) {
      console.error(error)
    }
```

Then yes, this does not currently work with Comlink (but it used to!), because what you are passing as a first argument here is an _object_. I am not traversing every parameter for `Comlink.proxy`-wrapped values because it makes performance completely unpredictable.

Either proxy the entire object or move the callback to its own parameter.

HTH.

(Side note: Don't use `comlink@alpha`, it's outdated.)

**Comment by Nithanaroy** [comment]
Thanks! Totally makes sense. And yeah I moved the callback as its own param.

This is an excellent library. I was able to get it working **easily** for neural network training in browser using tensorflow.js in browser. The workers do the calculations while the main thread takes care of visualizations while training is going on keeping the user engaged. 

1. main.js - https://github.com/Nithanaroy/tfjs-experiments/blob/master/3-mnist-cnn/model.js
2. worker.js - https://github.com/Nithanaroy/tfjs-experiments/blob/master/3-mnist-cnn/model-worker.js
